### DISCUSSION TITLE
xray 怎么查看 running config？

### DISCUSSION THREAD

**Discussion by lesca** [XTLS/Xray-core#2289] [General]
Hello,
运行 `xray run -confdir config/` 一切正常。

转到客户端，需要 `config.json` 因为不支持 `-confdir` 

所以怎么才能查看或者导出当前正在运行的配置文件，成为一个可用的 `config.json` 文件呢？

谢谢！

**Comment by yomnxkcs** [comment]
恭喜你发现了盲点！答案是没有这个功能。
多配置文件合并后的配置长什么样只能靠猜，没有输出合并后的配置功能，对debug配置很不友好。以前在v2ray那边提过issue不过没人理。
说到这忍不住提一嘴API的设计也很神奇！Inbound/Outbound的增删改查他只提供增删，改可以通过增删组合实现所以不提供也可以理解，但不提供查询功能就很让人摸不着头脑。再精确的时钟也要偶尔和外界校对一下时间吧，他就是不提供这个校对功能。

**Reply by shink** [reply]
So 请问下现在有这个功能了嘛 XD

**Reply by yomnxkcs** [reply]
现在只有一半。[多文件配置](https://xtls.github.io/config/features/multiple.html)的文档最后面写可以用`-dump`选项查看合并后的配置。但是只能看不能传给core。要是像以前那样可以输出成protobuf就好了。
